pennylane/qaoa/cost.py

```diff
@@ -159,15 +173,22 @@ def edge_driver(graph, reward):
             "'reward' cannot contain either '10' or '01', must contain neither or both."
         )
 
-    if not isinstance(graph, nx.Graph):
-        raise ValueError(f"Input graph must be a nx.Graph, got {type(graph).__name__}")
+    if not isinstance(graph, (nx.Graph, rx.PyGraph)):
+        raise ValueError(
+            f"Input graph must be a nx.Graph or rx.PyGraph, got {type(graph).__name__}"
+        )
 
     coeffs = []
     ops = []
 
+    is_rx = isinstance(graph, rx.PyGraph)
+    graph_nodes = graph.nodes()
+    graph_edges = sorted(graph.edge_list()) if is_rx else graph.edges
+    get_nvalue = lambda i: graph_nodes[i] if is_rx else i
```

Comment:
really nice that this works for both!

Reply:
just so I understand, the `get_nvalue` is because `PyGraph` nodes have to be indices, and can't be objects?

Might be good to add a comment here for future developers